File: 이것이_취업을_위한_코딩_테스트다/ch3 Greedy/number_card.py
# N, M를 공백을 기준으로 구분하여 입력 받기
n, m = map(int, input().split())
result=0
# i는 0부터 n-1 까지
for i in range(n) :
    #한줄 입력 받아서 리스트로 변환(m개 입력)
    data=list(map(int,input().split()))
    #리스트 중 가장 작은 값
    min_row = min(data)
    #작은 수 중 제일 큰 값 찾기
    result=max(result,min_row)
print(result) #결과 출력

File: 이것이_취업을_위한_코딩_테스트다/ch3 Greedy/NumberCard.java
import java.lang.reflect.Array;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;

public class NumberCard {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in); // 입력 받을 객체
        int m = sc.nextInt();
        int n = sc.nextInt();
        int[][] arr = new int[n][m];
        int result = Integer.MIN_VALUE; // 정수 중 가장 최솟값
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                arr[i][j] = sc.nextInt();
                // 각 행의 최솟값을 찾는 방법
                int min = Arrays.stream(arr[i]).min().getAsInt();
                // 각 행의 최솟값들 비교,
                result = Math.max(result, min);
            }
        }
        System.out.println(result);
        sc.close();
    }
}
